docs/content/docs/deployment/memory/network_mem_tuning.md

```diff
@@ -0,0 +1,176 @@
+
+* `taskmanager.network.memory.buffer-debloat.samples` - This adjusts the number of samples over which throughput measurements are averaged out. The frequency of the collected samples can be adjusted via `taskmanager.network.memory.buffer-debloat.period`. The fewer the samples, the faster the reaction time of the debloating mechanism, but a higher chance of a sudden spike or drop of the throughput which can cause the buffer debloating mechanism to miscalculate the optimal amount of in-flight data.
+
+* `taskmanager.network.memory.buffer-debloat.threshold-percentages` - An optimization for preventing frequent buffer size changes (i.e. if the new size is not much different compared to the old size).
+
+Consult the [configuration]({{< ref "docs/deployment/config" >}}#full-taskmanageroptions) documentation for more details and additional parameters.
+
+Here are [metrics]({{< ref "docs/ops/metrics" >}}#io) you can use to monitor the current buffer size:
+* `estimatedTimeToConsumerBuffersMs` - total time to consume data from all input channels
+* `debloatedBufferSize` - current buffer size
+
+### Limitations
+
+Currently, there are a couple of cases that are not handled automatically by the buffer debloating feature.
+
+#### Large records
+
+If your record size exceeds [minimum memory segment size]({{< ref "docs/deployment/config" >}}#taskmanager-memory-min-segment-size), buffer debloating
+can shrink the buffer size so much, that network stack will require more than one buffer to transfer a
+single record. This can have adverse effects on the throughput, without actually reducing the amount of the in-flight data. However this might be not an issue for you.
+
+#### Multiple inputs and unions
+
+Currently, the throughput calculation and buffer debloating happens on the subtask level. If your subtask has
+multiple different inputs, or it has single but unioned input, this can cause some problems. It might be especially
+visible if those different inputs have vastly different throughputs. In such case as the result of buffer debloating
+low throughput input might have still too much buffered in-flight data, while the hight throughput input might
+have too small buffers to sustain that throughput. We recommend paying special attention to such subtasks
+when testing this feature.
+
+#### Buffer size and number of buffers
+
+Currently, buffer debloating only caps the maximal used buffer size. The actual buffer size and the number of
+buffers remains unchanged. One consequence of this is that the debloating can not reduce memory usage of
+your job. To achieve that, you have to manually reduce either the amount or the size of the buffers. Secondly
+if you want to further reduce the amount of buffered in-flight data below what buffer debloating allows currently,
+you still might want to manually reduce the configured number of buffers.
```

Comment:
```suggestion
### Limitations

Currently, there are a few cases that are not handled automatically by the buffer debloating mechanism.

#### Large records

If your record size exceeds the [minimum memory segment size]({{< ref "docs/deployment/config" >}}#taskmanager-memory-min-segment-size), buffer debloating can potentially shrink the buffer size so much, that the network stack will require more than one buffer to transfer a single record. This can have adverse effects on the throughput, without actually reducing the amount of in-flight data. 

#### Multiple inputs and unions

Currently, the throughput calculation and buffer debloating happen on the subtask level. 

If your subtask has multiple different inputs or it has a single but unioned input, buffer debloating can cause the input of the low throughput to have too much buffered in-flight data, while the input of the high throughput might have buffers that are too small to sustain that throughput. This might be particularly visible if the different inputs have vastly different throughputs. We recommend paying special attention to such subtasks when testing this feature.

#### Buffer size and number of buffers

Currently, buffer debloating only caps at the maximal used buffer size. The actual buffer size and the number of buffers remain unchanged. This means that the debloating mechanism cannot reduce the memory usage of your job. You would have to manually reduce either the amount or the size of the buffers. 

Furthermore, if you want to reduce the amount of buffered in-flight data below what buffer debloating currently allows, you might want to manually configure the number of buffers.
```